File: AdoptaMe_Spring/AdoptaMe/src/main/java/mx/edu/unpa/apidroid/domains/Usuario.java
package mx.edu.unpa.apidroid.domains;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "usuario")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idUsuario;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(nullable = false, length = 100)
    private String apellidoPaterno;

    @Column(length = 100)
    private String apellidoMaterno;

    @Column(nullable = false, unique = true, length = 150)
    private String email;

    @Column(nullable = false, length = 255)
    private String password;

    @Column(length = 20)
    private String telefono;

    @Builder.Default
    @Column(nullable = false)
    private Boolean activo = true;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime fechaRegistro;
}

File: AdoptaMe_Android/MiAndroid/app/src/main/java/mx/edu/unpa/miandroid/model/Usuario.kt
package mx.edu.unpa.miandroid.model

import java.util.Date

data class Usuario(
    var id:Int?=null,
    var nombre: String,
    var apellidoPaterno: String,
    var apellidoMaterno: String?=null,
    var email: String,
    var telefono: String?=null,
    var password: String?=null,
    var foto: String?=null,
    var activo: Boolean=true,
    var fechaRegistro: String?=null
)
